### ISSUE TITLE
Link menus to the 'Configure' tab

### ISSUE THREAD

**Issue by ghost** [alexfinnarn/backdrop-xml-sitemap#54]
When you're at `/admin/config/search/xmlsitemap/settings` and click on a menu name under the 'Menu link' vertical tab, it takes you to menu's 'Edit links' tab (e.g. `/admin/structure/menu/manage/main-menu/edit`). Since you want to configure XML Sitemap, it should take you to the 'Configure' tab instead (e.g. `/admin/structure/menu/manage/main-menu/configure`).

**Comment by alexfinnarn** [comment]
So that link is gathered from `entity_get_info()`, however, the xmlsitemap_menu module already alters the "menu_link" entity info. 

I've made a PR that change the links to go to "Configure", but I'm not sure that should be done as all other code depending on that link might break. 

This issue also brings up a regression introduced by fixing #42. I placed the code for that into this PR, but the other modules need to be checked to make sure that they declare a hook to be included on the settings page.

**Comment by jenlampton** [comment]
> that link is gathered from entity_get_info() ... all other code depending on that link might break.

Can't we just update the link to configure for this instance, without changing something that affects anything else?

> the other modules need to be checked to make sure that they declare a hook to be included on the settings page.

Do we know if there are any "other modules" yet, and if so, what they are?

**Comment by alexfinnarn** [comment]
> Do we know if there are any "other modules" yet, and if so, what they are?

That was for declaring `hook_xmlsitemap_link_info()` which was actually merged in for the xmlsitemap_menu module in another PR so I've removed that part. Looks like the modules that should be in that list are there.

<img width="484" alt="screen shot 2019-01-21 at 12 31 50 pm" src="https://user-images.githubusercontent.com/3640707/51490245-aa2f0500-1d78-11e9-8675-e39ef8652993.png">

> Can't we just update the link to configure for this instance, without changing something that affects anything else?

You could use a form alter but the structure of the array for those menu links gets icky and is easiest to do in the `xmlsitemap_menu_entity_info_alter()` function like in the attached PR. I tried for a bit to make it work and someone else could finish it on the `feature/54-a` branch below, but it looks a bit brittle for my tastes to only save the user from an additional click.  

https://github.com/backdrop-contrib/xmlsitemap/compare/feature/54-a?expand=1

Unless someone wants to finish that code in a bulletproof way, I'm leaning toward closing this issue in order to not cause regressions in the future due to brittle code.

**Comment by quicksketch** [comment]
I think this issue may be simpler than it appears. The code in `xmlsitemap_menu_entity_info_alter()` is almost unaltered from what it was in the Drupal 7 module. It's not taking into account that the menu path for configuring menus (and many other entity types, like node types and taxonomy vocabularies) was changed from `/edit` to `/configure` in https://github.com/backdrop/backdrop-issues/issues/771.

Right now, XML Sitemap is pointing to a _non-existent_ path. There is nothing at `admin/structure/menu/manage/main-menu/edit` at all, and due to the way the menu system handles missing sub-pages, it just strips off the `/edit` and acts as though it wasn't there at all. You end up getting the listing page (the default local task) instead of the configuration form (which is now at `/configure`).

I think all that needs to be done here is updating the path. I filed a simple PR at https://github.com/backdrop-contrib/xmlsitemap/pull/108.
